### PR TITLE
DOC: Update compiler guidance for Windows

### PR DESCRIPTION
**Pull request by jamesmyatt**
... to make it a bit more encouraging.

### CODE REVIEW

File: doc/source/contributing.rst [hunk 1]
```diff
@@ -125,21 +125,24 @@ requires a C compiler and Python environment. If you're making documentation
...
-<https://docs.python.org/devguide/setup.html#build-dependencies>`_ for getting a
+platform you're using. Follow the `CPython contributing guide
+<https://devguide.python.org/setup/#compile-and-build>`_ for getting a
 compiler installed. You don't need to do any of the ``./configure`` or ``make``
 steps; you only need to install the compiler.
 
 For Windows developers, the following links may be helpful.
 
+* **TL;DR** (Python 3.5+) install `Visual Studio 2017 <https://visualstudio.com/>`_, select
+  the **Python development workload** and the **Python native development tools** option
```
[the remark this answers is not in the document]
Comment (jamesmyatt) [reply doc/source/contributing.rst]: Yeah, I wondered about that. But is it OK in principle?

**Review by chris-b1 — APPROVED** [review]

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: doc/source/contributing.rst [hunk 2]
```diff
@@ -125,21 +125,25 @@ requires a C compiler and Python environment. If you're making documentation
...
 extensions, which means you need a C compiler. This process depends on which
-platform you're using. Follow the `CPython contributing guidelines
-<https://docs.python.org/devguide/setup.html#build-dependencies>`_ for getting a
+platform you're using. Follow the `CPython contributing guide
+<https://devguide.python.org/setup/#compile-and-build>`_ for getting a
 compiler installed. You don't need to do any of the ``./configure`` or ``make``
 steps; you only need to install the compiler.
 
-For Windows developers, the following links may be helpful.
+For Windows developers, when using Python 3.5 and later, it is sufficient to
```
[the remark this answers is not in the document]
Comment (jreback) [reply doc/source/contributing.rst]: this should be >= 3.5 i think

[the remark this answers is not in the document]
Comment (jamesmyatt) [reply doc/source/contributing.rst]: The current version of the Python dev guide says 3.6+, but the MS Python engineering blog posts say 3.5+.

https://blogs.msdn.microsoft.com/pythonengineering/2017/03/07/python-support-in-vs2017/

[the remark this answers is not in the document]
Comment (datapythonista) [reply doc/source/contributing.rst]: may be @zooba can help?

[the remark this answers is not in the document]
Comment (chris-b1) [reply doc/source/contributing.rst]: So my understanding is that the VS 2015 _toolchain_ is required for python 3.5 extensions, but VS 2017 supports multiple toolchains, and the "Python native development tools" installs both the 2015/2017 c++ toolchains, so works for python 3.5+

[the remark this answers is not in the document]
Comment (zooba) [reply doc/source/contributing.rst]: That's correct, though IIRC Python 3.5 was never updated to automatically detect VS 2017, which is why the devguide refers to the newer version.

**Review by WillAyd — APPROVED** [review]
Fine with me @jreback

**Comment by jreback** [comment]
thanks @jamesmyatt

**Pull request merged**